### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by njmulsqb**
examples in accordance with changes to support objects instead of individual params

Signed-off-by: Najam Ul Saqib <najamulsaqib@tutamail.com>

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ const zaproxy = new ZapClient(zapOptions);
 Callbacks:
 
 ```js
-zaproxy.spider.scanAsUser(contextId, userId, sutBaseUrl, maxChildren, recurse, subtreeonly, (err, resp) => {
+zaproxy.spider.scanAsUser(argsObject, (err, resp) => {
```
Comment (thc202) [inline README.md:53]: Better pass the arguments as an object rather than replace them all.

Comment (njmulsqb) [reply README.md:53]: You mean like this `{contextId, userId, setBaseUrl, maxChildren, recurse, subtreeonly}`?

Comment (thc202) [reply README.md:53]: Yes.

Comment (thc202) [reply README.md:53]: (Tweaking to match the name of the API arguments though.)

**Comment by thc202** [comment]
This should be updated to match the latest changes.

File: README.md [hunk 2]
```diff
@@ -16,6 +16,7 @@
 </a>
 
 <br/><br/>
+
```
Comment (thc202) [inline README.md:19]: Why?

File: README.md [hunk 3]
```diff
@@ -29,41 +30,38 @@ npm install zaproxy
...
 ```js
 const ZapClient = require('zaproxy');
 
-const zapOptions = {
-  apiKey: <the key you supplied to ZAP when you started it>, // I.E. 'v90dnblpqs1pcac991tn2oudl'
-  proxy: <protocol>://<host>:<port> // I.E. 'http://192.168.0.10:8080'
-};
-
-const zaproxy = new ZapClient(zapOptions);
+async function invokeApi() {
```
Comment (thc202) [inline README.md:41]: I don't think this is necessary here.

File: README.md [hunk 4]
```diff
@@ -74,18 +72,15 @@ The Node API methods have the same signature as the API documentation, featuring
...
-**Callback mode**: If you provide a callback as the last parameter, the callback will be called with error and response arguments, with the response being an object that corresponds to the JSON output of the API call.  
-**Promise mode**: If you wish to use the modern approach, simply don't provide a callback, and a native promise will be returned for you to deal with as you wish.
-
 ## Getting Help
 
 For help using the OWASP ZAP API refer to:
 
-  * [API Documentation](https://www.zaproxy.org/docs/api/);
-  * [OWASP ZAP User Group](https://groups.google.com/group/zaproxy-users) - for asking questions;
+- [API Documentation](https://www.zaproxy.org/docs/api/);
```
Comment (thc202) [inline README.md:79]: Why these unnecessary changes?

File: README.md [hunk 5]
```diff
@@ -29,41 +30,38 @@ npm install zaproxy
...
-Promises:
-
-```js
-await zaproxy.spider.scanAsUser(contextId, userId, sutBaseUrl, maxChildren)
-  .then(
-    resp => console.log(JSON.stringify(resp)),
-    err => `Error occurred while attempting to scan as user. Error was: ${err.message}`
-  );
+let params = {
+  contextId,
```
Comment (thc202) [inline README.md:56]: Some of the parameters don't have the correct case, they should be either mapped or corrected.

File: README.md [hunk 6]
```diff
@@ -29,41 +30,38 @@ npm install zaproxy
...
+  userId,
+  setBaseUrl,
```
Comment (thc202) [inline README.md:58]: It was `sut` e.g. "site under test".

File: README.md [hunk 7]
```diff
@@ -29,41 +30,38 @@ npm install zaproxy
...
+  maxChildren,
+  recurse,
+  subtreeonly,
+};
+let response = await zaproxy.spider.scanAsUser(params);
```
Comment (thc202) [inline README.md:63]: If we are expected to `await` shouldn't the `await` in the function itself be dropped? Which would allow to handle the promise.

File: README.md [hunk 8]
```diff
@@ -74,18 +72,15 @@ The Node API methods have the same signature as the API documentation, featuring
...
 ## Getting Help
 
 For help using the OWASP ZAP API refer to:
 
-  * [API Documentation](https://www.zaproxy.org/docs/api/);
-  * [OWASP ZAP User Group](https://groups.google.com/group/zaproxy-users) - for asking questions;
+- [API Documentation](https://www.zaproxy.org/docs/api/);
+- [OWASP ZAP User Group](https://groups.google.com/group/zaproxy-users) - for asking questions;
+
+For specific help with this Node API, contact [@njmulsqb](https://github.com/njmulsqb) (the maintainer).
```
Comment (thc202) [inline README.md:82]: Better link to the user group.

File: README.md (hunk 7 above)
Comment (njmulsqb) [reply README.md:63]: Dropped in what sense? I didn't get you

File: README.md (hunk 3 above)
Comment (njmulsqb) [reply README.md:41]: It defines how the client can be utilized.

File: README.md (hunk 8 above)
Comment (njmulsqb) [reply README.md:82]: You mean I remove the maintainer line? I just updated the previous one

File: README.md (hunk 4 above)
Comment (njmulsqb) [reply README.md:79]: Prettier again :)

**Comment by njmulsqb** [comment]
Hi, can someone review this PR please? I am willing to release latest version of NodeJS API ASAP. Thank you

**Comment by njmulsqb** [comment]
Fixed :)

**Comment by kingthorin** [comment]
Thanks!

File: README.md (hunk 7 above)
Comment (thc202) [reply README.md:69]: Do not `await` in the functions. We should return the promise and let the user decided what to do: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises

File: README.md (hunk 8 above)
Comment (thc202) [reply README.md:88]: I mean link to the user group instead of a personal account.

File: README.md [hunk 9]
```diff
@@ -47,23 +57,17 @@ const zaproxy = new ZapClient(zapOptions);
...
 ```js
-await zaproxy.spider.scanAsUser(contextId, userId, sutBaseUrl, maxChildren)
-  .then(
-    resp => console.log(JSON.stringify(resp)),
-    err => `Error occurred while attempting to scan as user. Error was: ${err.message}`
-  );
+let params = {
+  contextid,
+  userid,
+  sutbaseurl,
```
Comment (thc202) [inline README.md:64]: The parameter needs to be mapped.

Comment (njmulsqb) [reply README.md:64]: Can you explain `mapped` a bit more?

File: README.md (hunk 2 above)
Comment (thc202) [reply README.md:19]: Better revert these changes, we don't want to do changes based on custom configs. If you want to enforce the format that should be done in a separate PR and CI should enforce it.

File: README.md (hunk 8 above)
Comment (thc202) [reply README.md]: I see now there's already the link to the group so yeah, it was to remove the line, sorry.

File: README.md (hunk 9 above)
Comment (njmulsqb) [reply README.md]: Done

**Comment by njmulsqb** [comment]
Are we ready to merge? @thc202

**Comment by njmulsqb** [comment]
Hey Team,

I see everyone is pretty busy with all making ZAP awesome with every passing day, I am also glad to be a small part of it. I just need little attention of the maintainers here, this PR is open for 3 months now, it would be great if we can merge this PR and release NodeJS API to ZAP users. 
I really want to take NodeJS API up to the standard of where our Python & Java APIs stand as of now. I truly believe that NodeJS API holds lots of potential since JS usage is rising up in industry so people are definitely going to prefer an updated ZAP API that is built with NodeJS. 
Once we have our NodeJS API ready, I can maybe write a blog post as well to explain the usage of it ;)

File: README.md (hunk 7 above)
Comment (thc202) [reply README.md]: This was not addressed, it needs to be done in the generated code not here.

File: README.md (hunk 2 above)
Comment (thc202) [reply README.md]: Not all changes were reverted.

File: README.md (hunk 9 above)
Comment (thc202) [reply README.md]: For the others they didn't need to be changed (though it will work fine as is).

File: README.md (hunk 7 above)
Comment (njmulsqb) [reply README.md]: Oh I thought generator's code has been finalized. No worries, let me fix it and create a PR

Comment (njmulsqb) [reply README.md]: https://github.com/zaproxy/zaproxy/pull/7854

File: README.md (hunk 2 above)
Comment (njmulsqb) [reply README.md]: Is there any easy way to see what changes are left now? I just tried to see through commit history but couldnt locate any more changes that need to be reverted.

Comment (thc202) [reply README.md]: See the diff in https://github.com/zaproxy/zap-api-nodejs/pull/28/files

Comment (njmulsqb) [reply README.md]: What about now?

File: README.md [hunk 10]
```diff
@@ -74,18 +73,13 @@ The Node API methods have the same signature as the API documentation, featuring
...
-**Promise mode**: If you wish to use the modern approach, simply don't provide a callback, and a native promise will be returned for you to deal with as you wish.
-
 ## Getting Help
 
 For help using the OWASP ZAP API refer to:
 
-  * [API Documentation](https://www.zaproxy.org/docs/api/);
-  * [OWASP ZAP User Group](https://groups.google.com/group/zaproxy-users) - for asking questions;
+-  * [API Documentation](https://www.zaproxy.org/docs/api/);
+-  * [OWASP ZAP User Group](https://groups.google.com/group/zaproxy-users) - for asking questions;
```
Comment (thc202) [inline README.md:81]:
```suggestion
  * [API Documentation](https://www.zaproxy.org/docs/api/);
  * [OWASP ZAP User Group](https://groups.google.com/group/zaproxy-users) - for asking questions;
```

File: README.md [hunk 11]
```diff
@@ -38,32 +38,31 @@ You can disable the API key when running ZAP if you are on a trusted network and
 const ZapClient = require('zaproxy');
 
 const zapOptions = {
-  apiKey: <the key you supplied to ZAP when you started it>, // I.E. 'v90dnblpqs1pcac991tn2oudl',
-  proxy: <protocol>://<host>:<port> // I.E. 'http://192.168.0.10:8080'
-};
-
-const zaproxy = new ZapClient(zapOptions);
-```
```
Comment (thc202) [inline README.md:41]:
```suggestion
```

File: README.md [hunk 12]
```diff
@@ -38,32 +38,31 @@ You can disable the API key when running ZAP if you are on a trusted network and
...
-
-Callbacks:
+    apiKey: 'eahhr6h6kal92j21gkcnhkp80t',
+    proxy: {
+      host: '127.0.0.1',
+      port: 8080,
+    },
+  };
+  
+  const zaproxy = new ZapClient(zapOptions);
```
Comment (thc202) [inline README.md:49]:
```suggestion
  apiKey: 'eahhr6h6kal92j21gkcnhkp80t',
  proxy: {
    host: '127.0.0.1',
    port: 8080,
  },
};

const zaproxy = new ZapClient(zapOptions);
```

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!
